Client/Frontend/Menu/MenuView.swift

```diff
@@ -298,7 +299,19 @@ class MenuView: UIView {
 
 
     private func layoutMenu() {
-        let menuHeight = self.collectionView(menuPagingView, layout: self.menuPagingLayout, sizeForItemAtIndexPath: NSIndexPath(forItem: 0, inSection: 0)).height
+        let numberOfItemsInRow = CGFloat(menuItemDataSource?.numberOfItemsPerRowInMenuView(self) ?? 0)
+        menuPagingLayout.maxNumberOfItemsPerPageRow = Int(numberOfItemsInRow)
+
+        // calculate cell size
+        var cellWidth = (self.bounds.size.width - itemPadding) / numberOfItemsInRow
+        cellWidth -= itemPadding
+        let cellHeight = CGFloat(menuRowHeight)
```

Comment:
I wonder if it's possible to move these calculations into the layout? Instead of using self.bounds, you'd use the connected collection view's bounds. I think you can also infer the height info from that too.